### PR TITLE
Handle case where cluster object has pure attribute

### LINKED ISSUE #31: Looks like auto scaling is not working for us
In theory our workers should scale up and down, but my recent tests do not see this happening in practice right now.

### PR DESCRIPTION
**Pull request by ocaisa**
Alternative to #35 also fixes #31

### CODE REVIEW

File: jobqueue_features/decorators.py [hunk 1]
```diff
@@ -173,6 +173,11 @@ def _get_cluster_attribute(self, cluster, attribute, default, **kwargs):
     def _submit(self, cluster, client, f, *args, **kwargs):
         # For MPI tasks, let's assume functions are not pure (by default)
         pure, kwargs = self._get_cluster_attribute(cluster, "pure", False, **kwargs)
+        # Handle the case where the cluster had a pure attribute
+        if hasattr(cluster, "pure") and pure is None:
```
Comment (ocaisa) [inline jobqueue_features/decorators.py:177]: Maybe I should be checking the value of `cluster.pure` rather than assuming it was `None`

Comment (AdamWlodarczyk) [reply jobqueue_features/decorators.py:177]: So are we updating the cluster attributes from the task? Should we set the `pure` attribute back to previous at the end of the task? 

Also - we could go for ~`cluster.pure = bool(getattr(cluster, "pure", False))`~ `cluster.pure = bool(kwargs.get("pure"))` I think.

Comment (ocaisa) [reply jobqueue_features/decorators.py:177]: Hmm, you are right, maybe submit is not the right place to set this value at all. We should be doing it when the cluster is declared.

Comment (ocaisa) [reply jobqueue_features/decorators.py:177]: and the value set at that time should persist but can be overridden by a kwarg.

**Pull request closed without merging**